src/api/common_api.js

```diff
@@ -138,6 +142,126 @@ module.exports = {
             }
         },
 
+        // bucket lifecycle
+
+        bucket_lifecycle_rule_expiration: {
+            type: 'object',
+            properties: {
+                days: {
+                    minimum: 1,
+                    type: 'integer'
+                },
+                date: {
+                    idate: true
+                },
+                expired_object_delete_marker: {
+                    type: 'boolean'
+                }
+            }
+        },
+        bucket_lifecycle_rule_status: {
+            enum: ['Enabled', 'Disabled'],
+            type: 'string'
+        },
+        bucket_lifecycle_rule_and_operator: {
+            type: 'object',
+            properties: {
+                prefix: {
+                    type: 'string'
+                },
+                tags: {
+                    type: 'array',
+                    items: {
+                        $ref: '#/definitions/tag',
+                    },
+                },
+                object_size_greater_than: {
+                    type: 'number',
+                },
+                object_size_less_than: {
+                    type: 'number',
+                },
+            },
+        },
+        bucket_lifecycle_rule_filter: {
+            type: 'object',
+            properties: {
+                prefix: {
+                    type: 'string'
+                },
+                tag: {
+                    $ref: '#/definitions/tag'
+                },
+                object_size_greater_than: {
+                    type: 'number'
+                },
+                object_size_less_than: {
+                    type: 'number'
+                },
+                and: {
```

Comment:
we don't necessarily have to use the same structure as the REST API.
see the discussion here https://github.com/noobaa/noobaa-core/pull/6849#discussion_r790244522